test/blackbox-tests/test-cases/pkg/opam-package-files-unix-error.t

```diff
@@ -0,0 +1,49 @@
+This test demonstrates the behaviour when a Unix error is encountered when copying the
+files/ directory from a package directory inside an opam repostory.
+
+  $ . ./helpers.sh
+
+Generate a mock opam repository
+  $ mkdir -p mock-opam-repository
+  $ cat >mock-opam-repository/repo <<EOF
+  > opam-version: "2.0"
+  > EOF
+
+
+Make a package with a patch
+  $ mkpkg with-patch <<EOF
+  > opam-version: "2.0"
+  > EOF
+
+  $ fname1="foo.patch"
+  $ fname2="dir/bar.patch"
+  $ opam_repo="mock-opam-repository/packages/with-patch/with-patch.0.0.1"
+  $ mkdir -p $opam_repo/files/dir
+  $ cat >$opam_repo/files/$fname1 <<EOF
+  > foo
+  > EOF
+  $ cat >$opam_repo/files/$fname2 <<EOF
+  > bar
+  > EOF
+We remove the read permissions for dir/
+
+  $ chmod -r $opam_repo/files/dir
+
+The error message should have a location for the opam repository.
+
+This does not currently seem to be the case.
```

Comment:
This will be fixed once this todo is resolved:
https://github.com/ocaml/dune/blob/b982acf9a7fb52474720bc5257b812f824b87616/bin/pkg.ml#L368-L369

cc @Leonidas-from-XIV